flocker/node/agents/gce.py

```diff
@@ -654,8 +656,13 @@ def detach_volume(self, blockdevice_id):
 
                 if potentially_detaching_error is not None:
                     try:
+                        # We want to poll until _get_attached_to no longer
+                        # returns a compute_instance_id, and instead raises an
+                        # UnattachedVolume exception.
```

Comment:
Thanks yea, sorry about the original bug.
I guess this might be an argument for having a mock version of GCEOperations (so I could unit test this logic)... although the failure modes here don't affect production flocker in negative ways, just the cleanup code in our acceptance tests.